DummyServer/Controllers/BuildController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BuildNotifications.Plugin.DummyServer;
+using BuildNotifications.PluginInterfaces.Builds;
+using DummyServer.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DummyServer.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class BuildController : ControllerBase
+    {
+        private readonly IDataStorage _dataStorage;
+
+        public BuildController(IDataStorage dataStorage)
+        {
+            _dataStorage = dataStorage;
+        }
+
+        [HttpGet]
+        public IEnumerable<WebBuild> Get()
+        {
+            return _dataStorage.Builds().Select(b => new WebBuild(b));
+        }
+
+        [HttpGet]
+        [Route(nameof(Extern))]
+        public IEnumerable<Build> Extern()
+        {
+            return _dataStorage.Builds();
+        }
+
+        [HttpPost]
+        public void Post([FromBody] WebBuild webBuild)
+        {
+            var existingBuild = _dataStorage.Builds().FirstOrDefault(b => b.Id.Equals(webBuild.Id));
+            if (existingBuild == null)
+                _dataStorage.AddBuild(webBuild.BranchName, webBuild.DefinitionName, webBuild.UserName);
+            else
+            {
+                existingBuild.Status = webBuild.Status;
+                existingBuild.Reason = webBuild.Reason;
+                existingBuild.LastChangedTime = DateTime.Now;
+            }
+        }
+
+        [HttpDelete]
+        public void Delete(string id)
+        {
+            _dataStorage.DeleteBuild(id);
+        }
+
+        [HttpGet]
+        [Route(nameof(Permutate))]
+        public void Permutate() => _dataStorage.PermutateBuilds();
+
+        [HttpGet]
+        [Route(nameof(RandomizeBuildStatus))]
+        public void RandomizeBuildStatus() => _dataStorage.RandomizeBuildStatus();
+    }
+
+    public class WebBuild
```

Comment:
As far as I'm know, marking DTOs with the DataContractAttribute and its member with the DataMemberAttribute is considered good practice.

Reply:
That debatable https://social.msdn.microsoft.com/Forums/vstudio/en-US/39d97953-f3bd-4f52-8cc6-a99ffc6b16e7/is-it-a-good-practice-to-use-datacontract-attributes-on-business-entities?forum=wcf

But I added the attributes, because I personally don't have a strong opinion on them. A better question is, is it bad practice to use non-DTO objects like in Build/Definition and UserController?

Reply:
For now I'd say it's good enough for the dummy build server. So we can merge this PR now.